sway-core/src/semantic_analysis/cei_pattern_analysis.rs

```diff
@@ -0,0 +1,258 @@
+// CEI stands for "Checks, Effects, Interactions".  We check that no storage writes
```

Comment:
It turns out that we also want to check for storage reads here which include:
* Storage read intrinsics.
* `StorageAccess` expressions.
* `asm` blocks using `srw` or srwq`.
* `asm` blocks using [`bal`](https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/instruction_set.md#bal-balance-of-contract-id) because `bal` actually reads from storage.

Reply:
Great, I'll add these checks as a separate type of warning, so the programmer could, in principle, silence it independently of the modification-after-call warnings.